src/Workspaces/CoreTest/ShadowCopyAndAllowRedirectsAssemblyLoaderTests.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#if !NETCOREAPP
+
+using System.Reflection;
+using System.Text;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.UnitTests
+{
+    [CollectionDefinition(Name)]
+    public class AssemblyLoadTestFixtureCollection : ICollectionFixture<AssemblyLoadTestFixture>
+    {
+        public const string Name = nameof(AssemblyLoadTestFixtureCollection);
+        private AssemblyLoadTestFixtureCollection() { }
+    }
+
+    [Collection(AssemblyLoadTestFixtureCollection.Name)]
+    public sealed class ShadowCopyAndAllowRedirectsAssemblyLoaderTests : TestBase
+    {
+        private readonly AssemblyLoadTestFixture _testFixture;
+
+        public ShadowCopyAndAllowRedirectsAssemblyLoaderTests(AssemblyLoadTestFixture testFixture)
+        {
+            _testFixture = testFixture;
+        }
+
+        [Fact]
+        public void AssemblyLoading_MultipleVersions()
+        {
+            var sb = new StringBuilder();
+
+            var loader = new ShadowCopyAndAllowRedirectsAssemblyLoader(baseDirectory: null);
+            loader.AddDependencyLocation(_testFixture.Gamma.Path);
+            loader.AddDependencyLocation(_testFixture.Epsilon.Path);
+            loader.AddDependencyLocation(_testFixture.Delta2.Path);
```

Comment:
I was surprised that Delta1 wasn't included via 'AddDependencyLocation' here. Isn't that necessary for us to be put in a position to have to choose between Delta1 and Delta2?

Reply:
Delta1 isn't here because there is no Delta1 being shipped. In an actual source generator package, how would you ship two versions of the same DLL?

Reply:
In some sense, that's the whole point of this PR: there's no actual way to make an analyzer package do what the other test does.

Reply:
> Delta1 isn't here because there is no Delta1 being shipped. In an actual source generator package, how would you ship two versions of the same DLL?

I had the same question when I read the test. Please add a comment here :)

Reply:
Comment clarified.